### PR TITLE
Adds support for 1.13 and upgrades deps

### PR DESCRIPTION
**Pull request by jasonmit**
* Updated deps
* Added support for > 1.11 since descriptor logic changed
* Various clean up

Would like this merged before I bring over https://github.com/jasonmit/ember-cli-moment-shim/pull/6/files

### CODE REVIEW

File: tests/unit/helpers/ago-test.js [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 module('AgoHelper');
 
-var FAKE_HANDLEBARS_CONTEXT = {};
+let FAKE_HANDLEBARS_CONTEXT = {};
+let threeDaysAgo = new Date();
+threeDaysAgo.setDate(threeDaysAgo.getDate() - 3);
 
-test('one arg (date)', function() {
-  equal(callHelper(ago, [date(0), FAKE_HANDLEBARS_CONTEXT]), "45 years ago");
+test('one arg (date)', () => {
```
Comment (jasonmit) [inline tests/unit/helpers/ago-test.js]: Test was fragile, could be 45 years or 46 years depending on environment test is run in

**Comment by jasonmit** [comment]
ember-beta and canary are failing.. which is expected.  further changes are needed to support these

File: tests/unit/helpers/ago-test.js (hunk 1 above)
Comment (stefanpenner) [inline tests/unit/helpers/ago-test.js]: thanks

**Comment by stefanpenner** [comment]
we should have different ember try's working in different travis matrix entries. This allows us to mark canary as allowed failure

**Comment by jasonmit** [comment]
@stefanpenner all done with the env matrix.  I'll merge but will hold off on versioning until I finish up the moment-shim work

**Pull request merged**